lib/master.js

```diff
@@ -195,9 +195,35 @@ Master.prototype._startWorkers = function(remainingWorkers) {
                 type: 'config',
                 body: yaml.dump(self.config)
             });
+
+            var workerExit = function(code) {
```

Comment:
@Pchelolo one possible idea would be to have a `this._isStartup` flag which is set to `true` only on master process start and upon receiving `SIGHUP`. If that flag is active, the `cluster.on('exit')` handler becomes a no-op. Then, once `remainingWorkers` reaches `0`, the flag is set to `false`.